service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -29,6 +28,103 @@
  */
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
+
+    @ApiMayChange
+    default <T extends Service> LagomServiceBuilder serverFor(Class<T> serviceInterface, Class<? extends T> serviceImplementation){
+        return new LagomServiceBuilder(serviceBinding(serviceInterface, serviceImplementation), this);
+    }
+
+    @ApiMayChange
+    default <T extends Service> LagomServiceBuilder serverFor(Class<T> serviceInterface, T serviceInstance){
+        return new LagomServiceBuilder(serviceBinding(serviceInterface, serviceInstance), this);
+    }
+
+    class LagomServiceBuilder {
```

Comment:
I initially took the path of a ServiceBuilder that the user could use to bind the service and add the extra routers, like in:

```java
serverFor(Foo.class, FooImpl.class)
  .plusRouter(someOtherRouter)
  .bind()
```

Unfortunately, that won't work because an router usually need some other classes inject to it. 

Therefore I move to an approach in which the user can define `AdditionalRouters` impl together with a `Provider` to let Guice do the injection.

I also found it useful to have it all bound together instead of having some `List<Router>` to be defined somewhere. That was more complicated because we will need a Guice overrides thing like in Play. By default, we will need to provide an empty list that the user can fill, eventually.

The builder approach used here does the same, without needing to have a Guice overrides function. And we bring everything together, we bind the service and we add the extra routers. I see this as an advantage. And it has the bonus to be discoverable.

Reply:
What do you think about:

```java
bindRouters(
   bindRouter(OneRouter),
   bindRouter(AnotherRouter),
   bindRouter(OneLastRouter)
)
```
?

This is similar to the (now deprecated) [`bindServices`](https://www.lagomframework.com/documentation/1.2.x/java/ServiceImplementation.html). That would internally create the binding to `AdditionalRouters` (or default to `EmptyAdditionalRouters`).

Finally, `bindService` (which would still be compulsory) would internally invoke `binder.bind(LagomServiceRouter.class).to(JavadslServicesRouter.class);`


---------------------------------------

With the approach above, the `Module` code would look like;

```java
    bindService(HelloService.class, HelloServiceImpl.class);
    bindAdditionalRouters(
       bindRouter(OneRouter),
       bindRouter(AnotherRouter),
       bindRouter(OneLastRouter)
    )
    // other custom binds the user must provide to completely build the routers above
```

And no extra `Provider` must be implemented by the user.

This approach doesn't introduce a custom DSL, and is (arguably) discoverable as it introduces a `bindAdditionalRouters` as a top-level method using the `bindXyz` naming convention which users are familiar with.

Also, prevents the need to deprecate `bindService`.

Reply:
I'm trying what @ignasi35 suggested, from on top of this PR

Reply:
I think that there is a misunderstanding here about what we should to make it work with Guice and be able to define the paths. 

The reason I was introducing a `Provider` was to have a place where we can configure the path and get things injected. Maybe that was not directly obvious from the examples I added, but the idea was to be able to write the following:

```java
class RouterProvider(DepsPing depsPing, DepsPong, depsPong) implements Provider<AdditionalRouters> {
        @Override
        public AdditionalRouters get() {
            return new AdditionalRouters() {
                @Override
                public List<Router> getRouters() {
                    return Arrays.asList(
                        new PingRouter(depsPing).withPrefix("/ping"),
                        new PongRouter(depsPong).withPrefix("/pong")
                    );
                }
            };
        }
    }
```

@ignasi35's suggestion to have a `bindAdditionalRouters` to hide the complexity introduced by the `Provider` won't allow DI. And we need it for gRPC routers (and most probably any other akka-http router, eg: cluter http management). 

@raboof's suggestion to have an additional list of `.class` do allow DI, but does not allow for path configuration. This is not needed for gRPC because the path is already fixed, but it's needed for other kind of routers.

@ktoso's [implementation](https://github.com/ktoso/lagom/blob/wip-routers/service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java#L173) won't work either. It's a mix of @ignasi35 and @raboof and it doesn't allow injection, but do allow path configuration. 

The bottom line is:
* any approach requiring an instance (@ktoso/@ignasi35) allows path configuration, but blocks DI.
* any approach requiring a `.class` (@raboof) allows DI, but blocks path configuration.

Reply:
I'll be honest about the fact that I'm not getting the core of this all, it's still over my head somehow... 🤔 Is this a task you could continue on before the upcoming vacation?